PennyLaneAI/pennylane-lightning
# Add clang-tidy checks, runtime and compile time information, a C++ kernel test

### Before submitting

Please complete the following checklist when submitting a PR:

- [x] All new features must include a unit test.
      If you've fixed a bug or added code that should be tested, add a test to the
      [`tests`](../tests) directory!

- [x] All new functions and code must be clearly commented and documented.
      If you do make documentation changes, make sure that the docs build and
      render correctly by running `make docs`.

- [x] Ensure that the test suite passes, by running `make test`.

- [x] Add a new entry to the `.github/CHANGELOG.md` file, summarizing the
      change, and including a link back to the PR.

- [x] Ensure that code is properly formatted by running `make format`. 

When all the above are checked, delete everything above the dashed
line and fill in the pull request template.

------------------------------------------------------------------------------------------------------------

**Context:** As PR #245 becomes too huge, splitting is required. This is the first part of it. 

**Description of the Change:** There are miscellaneous changes. Some are listed below:

* Update `.clang-tidy` and source code accordingly.
* Documents now only use header files to suppress lots of clutters.
* Python files in `doc` are formatted.
* Add support to obtain compiled and runtime information through a binary file
* Add a test comparing gate operations between different kernels (`Test_GateImplementations_CompareKernels.cpp`)
* Fix a bug in `tests/test_measure.py` (a single qubit state is provided but `dev.probability` is called with two wires)

**Benefits:**

**Possible Drawbacks:**

**Related GitHub Issues:**
